Jenkinsfile

```diff
@@ -0,0 +1,69 @@
+#!groovy
+
+properties([
+    buildDiscarder(logRotator(daysToKeepStr: '20', numToKeepStr: '30')),
+
+    pipelineTriggers([pollSCM('H/15 * * * *')])
+])
+
+def test_ignition(ARCH, GOVERSION)
+{
+    node("${ARCH} && docker") {
+        def CGO = (ARCH == 'arm64') ? 1 : 0
+        withEnv(["TARGET=${ARCH}", "CGO_ENABLED=${CGO}",
+                 "GOARCH=${ARCH}", "GOVERSION=${GOVERSION}"]) {
+            stage("SCM $GOVERSION") {
+                checkout scm
+            }
+
+            stage("Build & Test $GOVERSION") {
+                sh '''#!/bin/bash -ex
+
+sudo chmod +x docker_build
+sudo chmod +x test
+sudo chmod +x build
+
+sed -i "s/_GOVERSION_/$GOVERSION/g" Dockerfile
```

Comment:
Most of our other projects that use this build image strategy maintain the build image Dockerfile in the repo and publish it separately. That way anyone (human or Jenkins alike) can use it to work or debug in the same environment. It also avoids the time or scripting of building it every time.